### ISSUE TITLE
Flaky span counts caused by flaky elasticsearch span queries

### ISSUE THREAD

**Issue by albertteoh** [jaegertracing/jaeger#2688]
**TLDR;** 
Indeterministic behaviour from ES span query results leading to flaky span counts (and span sets) for traces. This appears to happen when `ES_MAX_DOC_COUNT * num_index_shards < num_spans_in_trace`. Has anybody seen this before and know if there is an easy fix for it, perhaps in ES configuration? @pavolloffay 

**Observations**
The following gif demonstrates the symptom resulting from the flakey ES span query results. "Find Traces" is clicked multiple times in quick succession on the same query and the first trace's span count jumps between the following numbers: `[500, 490, 499, 489, 435]`.

![jaeger-ui-demo-flaky-spancount-es7](https://user-images.githubusercontent.com/26584478/101739419-9a12a100-3b1b-11eb-9221-f1458fc08f7e.gif)

Jaeger was ruled out as the cause of these flaky span counts by reproducing the flaky query results locally in ES, as this next gif demonstrates:

![postman-demo-flaky-span-query-es7](https://user-images.githubusercontent.com/26584478/101885751-a3b90900-3bee-11eb-9eb0-3478b717c28f.gif)

Note:
- The same query is executed multiple times in quick succession.
- The query is on a single shard (out of 5 shards): `?preference=_shards:2`.
- The query requests for a single document via the `size: 1` and `terminate_after: 1` query attributes.
- The spanID, start time and duration fields returned from the query differ between executions of the same query.
- I believe the indeterministic start times cause the flaky span counts due to how jaeger-query scrolls through the documents returned by ES: https://github.com/jaegertracing/jaeger/blob/master/plugin/storage/es/spanstore/reader.go#L388 
- This behaviour exists on both ES 6 and 7, and the gif above is from ES 7.

My suspicion is that [`TerminateAfter`](https://github.com/jaegertracing/jaeger/blob/6c2be456ca41cdb98ac4b81cb8d9a9a9044463cd/plugin/storage/es/spanstore/reader.go#L167) fetches the first `n` documents in-deterministically from each shard (before any sorting is applied), because removing `TerminateAfter` from the query resulted in correct span counts, though no longer limited.

**Comment by albertteoh** [comment]
Closing given no responses for some time. Have moved this question to Elasticsearch Issues so it's not lost: https://github.com/elastic/elasticsearch/issues/67215.
